### DISCUSSION TITLE
Poor performance Celery FastAPI

### DISCUSSION THREAD

**Discussion by manhtd98** [celery/celery#7630] [Q&A] [answered]
I use fastAPI+celery+rabbitmq but only can process 160ccu without worker processing. What's the stuck point?

[a comment by an automated account is not shown]

**Comment by manhtd98** [comment]
I can run code with 80 CCU. Don't know why that poor performance

**Comment by olii** [comment, marked as answer]
You need to share a code sample. We cannot help you without it. Keep in mind that FastAPI is async while celery is synchronous. Do you run celery code in a separate thread?

See: https://fastapi.tiangolo.com/tutorial/background-tasks/
